docs/oauth.md

```diff
@@ -0,0 +1,215 @@
+```
+
+```xml
+  connectionFields.xml Multiple Auth Types:
+
+  <field name="authentication" label="Authentication" category="authentication" value-type="selection" default-value="auth-user-pass" >
+    <selection-group>
+      <option value="auth-user-pass" label="Username and Password"/>
+      <option value="oauth" label="OAuth"/>
+    </selection-group>
+  </field>
+```
+
+In your connectionBuilder.js file you need to use your DB sepcific logic to handle how to pass in OAuth attributes. E.g. for snowflake:
+
+```js
+  connectionBuilder.js
+  if(authAttrValue == "oauth")
+    {
+        params["AUTHENTICATOR"] = "OAUTH";
+        params["TOKEN"] = attr["ACCESSTOKEN"];
+    }
+```
+
+# The OAuth Config File
+
+The OAuth Config file defines your oauth connector configs and also provide the ability to customize how your oauth flow should work.
+
+The OAuth Config file ([XSD](https://github.com/tableau/connector-plugin-sdk/blob/dev-2021.1/validation/oauth_config.xsd)) is indentified in the manifest using the `<oauth-config>` element. Here we discuss the structure of this file.
+
+## XML Elements
+
+### `<pluginOAuthConfig>`
+
+This is the parent element for all fields.
+
+### `Elements`
+
+Each OAuth config attribute is represented by an element in the XML, the element name is the attribute name and the content is the attribute value.
+
+| Name  | Type | Meaning | Optional? | Notes |
+| ----  | ------- | --------- | ----------- | ----------- |
+| dbclass | String | The dbclass for your oauthConfig | No | dbclass must be same with the `class` attribute in manifest.xml | 
+| clientIdDesktop | String | Client ID you registered for Tableau Desktop | Yes | This is not considered as secrets and will be stored in plain text | 
+| clienSecretDesktop | String | Client Secret you registered for Tableau Desktop | Yes | This is not considered as secrets and will be stored in plain text | 
+| redirectUrisDesktop | String[] | Redirect Urls for Desktop | Yes	| The host for redirectUrisDesktop must be a valid loopback address| 
+| authUri | String | Authorization endpoint URI | No | |
+| tokenUri | String | Token endpoint URI | No | |
+| userInfoUri | String | User Info UrI | Yes | |
+| instanceUrlValidationRegex | String | Use to validate against your OAuth instance Url. | Yes | |
+| scopes | String[] | scopes | No | |
+| capabilities | Map<String, String> | This defines how oauth flow behaves differently according the caps you set. | Yes | |
+| accessTokenResponseMaps | Map<String, String> | Key value pair that maps an initial token request response attribute <value> to Tableau recognized attribute <key> | No | |
+| refreshTokenResponseMaps | Map<String, String> | Key value pair that maps an refresh token request response attribute <value> to Tableau recognized attribute <key> | Yes | if not defined will use accessTokenResponseMaps by default |
+
+## Supported Capabilities
```

Comment:
```suggestion
## Supported OAuth Capabilities
```

Reply:
You may also want to add a line in the general capabilities page saying something like "For Oauth-specific capabilities, go here" and link back to this page.

Reply:
Yep sounds good.